### PR TITLE
JS: Detect insecure downloads that are saved to sensitive files.

### PR DESCRIPTION
**Pull request by erik-krogh**
Implements this suggestion: https://github.com/github/codeql/pull/3689#issuecomment-644854388

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/security/dataflow/InsecureDownloadCustomizations.qll [hunk 1]
```diff
@@ -13,7 +13,12 @@ module InsecureDownload {
   /**
    * A data flow source for download of sensitive file through insecure connection.
    */
-  abstract class Source extends DataFlow::Node { }
+  abstract class Source extends DataFlow::Node {
+    /**
+     * Gets a flow-label for this source
```
Comment (intrigus-lgtm) [inline javascript/ql/src/semmle/javascript/security/dataflow/InsecureDownloadCustomizations.qll:18]: Nit: Missing period.

**Comment by erik-krogh** [comment]
[Evaluation was uneventful](https://git.semmle.com/erik/dist-compare-reports/tree/profiling-erik-krogh.northeurope.cloudapp.azure.com_1593363883317)

File: javascript/ql/src/semmle/javascript/security/dataflow/InsecureDownloadCustomizations.qll [hunk 2]
```diff
@@ -58,13 +102,48 @@ module InsecureDownload {
...
    */
   class ClientRequestURL extends Sink {
     ClientRequest request;
 
     ClientRequestURL() { this = request.getUrl() }
 
     override DataFlow::Node getDownloadCall() { result = request }
+
+    override DataFlow::FlowLabel getALabel() {
+      result instanceof Label::SensitiveInsecureURL // TODO: Also non-sensitive.
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/security/dataflow/InsecureDownloadCustomizations.qll:115]: Is this TODO comment still needed? It looks like non-sensitive URLs are handled below
```suggestion
      result instanceof Label::SensitiveInsecureURL
```

Comment (erik-krogh) [reply javascript/ql/src/semmle/javascript/security/dataflow/InsecureDownloadCustomizations.qll:115]: Correct. I forgot to remove that comment after I implemented that.

**Pull request merged**